src/network/network.cpp

```diff
@@ -688,16 +695,13 @@ class TCPClientConnecter : TCPConnecter {
 
 
 /* Used by clients, to connect to a server */
-void NetworkClientConnectGame(const char *hostname, uint16 port, CompanyID join_as, const char *join_server_password, const char *join_company_password)
+void NetworkClientConnectGame(NetworkAddress &address, CompanyID join_as, const char *join_server_password, const char *join_company_password)
```

Comment:
const? (and also others)

Reply:
`NetworkAddress` currently is very special, and can modify itself. So it cannot be `const` atm.

But as this is not the first time being asked this, I will see what I can do after STUN to make `NetworkAddress` a bit less .. special :) `mutable` is what @LordAro suggests.